kubectl-cloudflow/cmd/deploy_command.go

```diff
@@ -261,8 +251,8 @@ func updateImageRefsWithDigests(spec cfapp.CloudflowApplicationSpec, pulledImage
 	return spec
 }
 
-func handleAuth(k8sClient *kubernetes.Clientset, namespace string, opts *deployOptions, firstPulledImage *dockerclient.PulledImage) {
-	dockerRegistryURL := firstPulledImage.DockerRegistryURL
+func handleAuth(k8sClient *kubernetes.Clientset, namespace string, opts *deployOptions, imageReference *dockerclient.ImageReference) {
+	dockerRegistryURL := imageReference.Registry
```

Comment:
Also change the LogExit here:
```
printutil.LogAndExit("Please provide username and password, by using both --username and --password-stdin, or, by using both --username and --password, or omit these flags to get prompted for username and password.")
``` To indicate the option to use no-registry-credentials, and explain when that makes sense.

Reply:
Added a text about the no credentials flag.